pandas/io/formats/format.py

```diff
@@ -1276,30 +1276,41 @@ def format_array(
     List[str]
     """
     fmt_klass: type[GenericArrayFormatter]
-    if is_datetime64_dtype(values.dtype):
+
+    if space is None:
+        space = get_option("display.column_space")
+
+    if float_format is None:
+        float_format = get_option("display.float_format")
+
+    if digits is None:
+        digits = get_option("display.precision")
+
+    if is_extension_array_dtype(values):
+        return values._format_array(
+            formatter,
+            float_format,
+            na_rep,
+            digits,
+            space,
+            justify,
+            decimal,
+            leading_space,
+            quoting,
+        )
+    elif is_datetime64_dtype(values.dtype):
```

Comment:
I should check to see if these datetlike are actually hit anymore.

Reply:
yah.  i expect that if you use `isinstance(values, ExtensionArray)` instead of `is_extension_array_dtype` above you can get the dt64 and td64 cases at the same time